### PR TITLE
cdp-listend: Factor out calls to notification handlers

### PR DESCRIPTION
**Pull request by jrha**
* Factor out calls to notification handlers to a new subroutine as much of the code is common.
* Log unknown notification types.
* Improve slightly ambiguous log messages regarding timing.
* Make it easier to support additional notification types in future.

### CODE REVIEW

**Comment by jrha** [comment]
@stdweird any thoughts?

File: src/main/scripts/cdp-listend [hunk 1]
```diff
@@ -192,25 +192,25 @@ while($ip1->recv($newmsg,1024)) {
...
+        call_handler($fetch, "--profile-time=$time", $delay);
     } elsif ($ntype eq "cdb") {
-        my $smear = int(rand($nch_smear));
-        logit('info', "$nch will be called in $smear seconds");
-        sleep($smear);
-        logit('info', "Calling $nch with unix time $time (after $smear seconds)");
-        system ("$nch") == 0
-            or logit('err',"[ERROR] call of cdbsync-nch ($nch) failed: $!");
+        my $delay = int(rand($nch_smear));
+        call_handler($nch, "", $delay);
```
Comment (ned21) [inline src/main/scripts/cdp-listend:200]: This is a bit ugly and technically `system` wants a list here not a scalar. I think a purist approach would be to make call_handler take an array_ref as the first arg, which would then use `system(@args)` form of system, but for the two use cases we have here I think that's overkill. 

So I'm in favour of the current implementation on the basis it's simple and easy to understand and thus more likely to work correctly than something that is more syntactical complex!

**Review by ned21 — APPROVED** [review]

File: src/main/scripts/cdp-listend [hunk 2]
```diff
@@ -192,25 +192,25 @@ while($ip1->recv($newmsg,1024)) {
...
+    } else {
+        logit('err', "[ERROR] Ignoring packet, unknown notification type");
     }
 }
 
+sub call_handler {
+    my ($handler, $handler_args, $delay) = @_;
+    logit('info', "$handler will be called in $delay seconds");
+    sleep($delay);
+    logit('info', "Calling $handler now");
```
Comment (ned21) [inline src/main/scripts/cdp-listend:210]: Just realised this drops the $time from the log message; does something else log the time of this invocation or are we losing the info about when the handler was actually called?

Comment (jrha) [reply src/main/scripts/cdp-listend:210]: It is still logged once when the notification packet is received, but as it is the embedded timestamp from the packet it isn't that helpful IMHO, e.g:

Previous behaviour:
```
Sep  8 00:02:02 example-host /usr/sbin/cdp-listend[4836]: Received UDP packet (ccm|1725750122) from 198.51.18.11
Sep  8 00:02:02 example-host /usr/sbin/cdp-listend[4836]: /usr/sbin/ccm-fetch will be called in 0 seconds (offset=0, smear=0)
Sep  8 00:02:02 example-host /usr/sbin/cdp-listend[4836]: Calling /usr/sbin/ccm-fetch with unix time 1725750122 (after 0 seconds)
```

New behaviour:
```
Sep  9 10:48:14 example-host /usr/sbin/cdp-listend[29011]: Received UDP packet (ccm|1725875294) from 198.51.18.11
Sep  9 10:48:14 example-host /usr/sbin/cdp-listend[29011]: /usr/sbin/ccm-fetch will be called in 0 seconds
Sep  9 10:48:14 example-host /usr/sbin/cdp-listend[29011]: Calling /usr/sbin/ccm-fetch now
```

**Pull request merged**